tests/components/synology_dsm/test_media_source.py

```diff
@@ -428,6 +428,8 @@ async def test_media_view(
 
     # success
     dsm_with_photos.photos.download_item = AsyncMock(return_value=b"xxxx")
-    tempfile.tempdir = tmp_path
-    result = await view.get(request, "mocked_syno_dsm_entry", "10_1298753/filename.jpg")
-    assert isinstance(result, web.Response)
+    with patch.object(tempfile, "tempdir", tmp_path):
```

Comment:
This is weird. Does this test actually save a file in a temporary directory? The library should just be patched instead and it's enough to assert that the library is called correctly. And of course we shouldn't create the view directly in the test but use our client to make a request to the server endpoint.

Reply:
The issue was that the test set the tempdir globally with `tempfile.tempdir = tmp_path`.
As I'm not familiar with the integration I fixed it in the least invasive way, and that was to patch it instead.

Reply:
Sure. This was just a side note.